docs/modules/ROOT/pages/adr/0033-event-based-billing-oddo.adoc

```diff
@@ -0,0 +1,233 @@
+====
+
+== Requirements
+
+. Reuse existing tools and/or components of AppCat to build a resilient event-based billing solution.
+. Enable resending data to Odoo (historical record retention is required).
+. Persist the state of each billing request.
+. Do not block service creation, deletion, or update due to errors in the billing system.
+. [[odoo-sync-state]] Provide a way to verify sync state between Odoo and the Kubernetes cluster.
+. Do not lose any event (created, deleted, scaled), as this has a direct financial impact.
+
+
+== Solution Options
+
+After careful evaluation, the two most promising solutions for implementing event-based billing are:
+
+. <<AppCat Controller>>
+. <<Runtime Library via Crossplane Composition Functions>>
+
+=== AppCat Controller
+
+Transitioning from metered to event-based billing requires leveraging Kubernetes controllers more extensively.
+Our existing AppCat controller already handles event forwarding and webhooks, making it a natural candidate for integrating billing logic.
+
+*Pros:*
+
+* Full control over the lifecycle of VSHN custom resource services.
+* Customizable retry logic.
+* Clear separation of concerns between billing and service management.
+* Flexible support for different persistence backends.
+
+*Cons:*
+
+* More difficult to block or revert service operations when billing events fail to post to Odoo.
+
+=== Runtime Library via Crossplane Composition Functions
+
+We can embed billing logic into our existing runtime library for Crossplane Composition Functions, thereby coupling service lifecycle events directly with billing logic.
+
+*Pros:*
+
+* Reconciliation happens directly during create, update, or delete operations.
+* Greater control over services when billing fails.
+* Billing logic is treated as a first-class part of provisioning/deprovisioning.
+
+*Cons:*
+
+* Persistence integration becomes more complex.
+* No clear separation between billing and service logic.
+* Unclear where billing actually occurs, which can reduce maintainability.
+* We can't actually react on deletion events. Crossplane doesn't propagate them to the functions.
+
+== Persistence Options
+
+Based on past experience, we anticipate the need to resend older data to Odoo due to potential issues either in AppCat or Odoo.
+A lightweight, reliable mechanism to store and replay billing data is essential.
+
+Required capabilities:
+
+* Filtering and querying historical events.
+* Manual replay.
+* Partial delivery of historical events.
+
+=== SQLite
+
+SQLite is a simple, embedded SQL database engine suitable for local persistence needs.
+
+*Pros:*
+
+* Minimal setup; no external infrastructure required.
+* Fast for local and sequential read/write operations.
+* Full SQL support.
+* ACID-compliant (supports WAL mode).
+* Self-contained `.db` file that's easy to handle and back up.
+* Supports pagination and filtering by retry state or timestamp.
+
+*Cons:*
```

Comment:
While I agree with the cons, they barely reflect the requirements defined. But fine by me to leave it like this.